pkg/drivers/kic/kic.go

```diff
@@ -92,6 +92,27 @@ func (d *Driver) Create() error {
 			ContainerPort: constants.DockerDaemonPort,
 		},
 	)
+
+	exists, err := oci.ContainerExists(d.OCIBinary, params.Name)
+	if err != nil {
+		glog.Warningf("failed to check if container already exists: %v", err)
+	}
+	if exists {
+		// if container was created by minikube it is safe to delete and recreate it.
+		if oci.IsCreatedByMinikube(d.OCIBinary, params.Name) {
+			glog.Info("Found already existing abandoned minikube container, will try to delete.")
+			if err := oci.DeleteContainer(d.OCIBinary, params.Name); err != nil {
+				glog.Errorf("Failed to delete a conflicting minikube container %s. You might need to restart your %s daemon and delete it manually and try again.", params.Name, params.OCIBinary)
+				return errors.Wrapf(err, "deleting already in-use mini-created container %s", params.Name)
+			}
+		} else {
+			// The conflicting container name was not created by minikube
+			// user has a container that conflicts with minikube profile name, will not delete users container.
+			glog.Errorf("You have a container named %s that conflicts with minikube profile name %s, please either remove manually or choose a different minikbue profile name", params.Name, params.Name)
```

Comment:
Log the error or return the error - not both. This results in duplicate errors being displayed.

Consider %q for the profile name for clarity. Also check the spelling of minikube.